dog_classifier/evaluate/evaluate_randomforest.py

```diff
@@ -150,16 +149,25 @@ def get_label_encoder(encoder_model):
     return encoder
 
 
-def create_confusion_matrix(label_encoder, y_true, y_pred):
+def create_confusion_matrix(label_encoder, n_classes, y_true, y_pred):
+    # I have to put this here, othwise I get a strange error:
+    # File "/home/beckstev/.local/anaconda3/lib/python3.7/site-packages/matplotlib/dviread.py", line 199, in __init__
+    # self.file = open(filename, 'rb')
+    # FileNotFoundError: [Errno 2] No such file or directory:
+    # '/home/user/.cache/matplotlib/tex.cache/4ee3b5cbb5a9a35df6cd457d83904876.dvi'
+    mpl.rcParams.update(mpl.rcParamsDefault)
+
     model_save_path = os.path.join(Path(os.path.abspath(__file__)).parents[2],
                                    "saved_models",
-                                   "autoencoder")
+                                   "autoencoder" + "_n_" + str(n_classes))
 
     cm = confusion_matrix(y_true, y_pred)
     cm = cm.astype('float') / cm.sum(axis=1)[:, np.newaxis]
 
     classes = label_encoder.inverse_transform(np.arange(max(y_true)+1))
-    if len(classes) == 120:
+    classes = [cl.replace('_', ' ') for cl in classes]
```

Comment:
Was passiert an dieser Stelle? Wieso ist das nötig?

Reply:
Because latex (at least my version) can't handle a underscore when it is not used in a math environment. Therefore, all underscores are replaced with blankspaces.